app/src/main/java/com/asfoundation/wallet/backup/use_cases/ShouldShowSystemNotificationUseCase.kt

```diff
@@ -2,29 +2,39 @@ package com.asfoundation.wallet.backup.use_cases
 
 import com.asfoundation.wallet.backup.repository.preferences.BackupSystemNotificationPreferences
 import com.asfoundation.wallet.repository.SharedPreferencesRepository
-import com.asfoundation.wallet.wallets.usecases.GetWalletInfoUseCase
+import com.asfoundation.wallet.wallets.domain.WalletInfo
+import io.reactivex.Single
+import java.util.concurrent.TimeUnit
 import javax.inject.Inject
 
 class ShouldShowSystemNotificationUseCase @Inject constructor(
-  private val getWalletInfoUseCase: GetWalletInfoUseCase,
   private val sharedPreferencesRepository: SharedPreferencesRepository,
   private val backupSystemNotificationPreferences: BackupSystemNotificationPreferences
 ) {
 
   companion object {
     private const val PURCHASE_NOTIFICATION_THRESHOLD = 2
+    private const val DISMISS_PERIOD = 30L
   }
 
-  operator fun invoke(walletAddress: String): Boolean {
+  operator fun invoke(walletInfo: WalletInfo): Single<Boolean> {
+    val meetsAllNotificationConditions = meetsLastDismissCondition(walletInfo.wallet) && meetsCountConditions(walletInfo.wallet)
+    return if (walletInfo.hasBackup.not()) {
```

Comment:
What about `Single.just(walletInfo.hasBackup.not() && meetsAllNotificationConditions)` ?